File: 100/125 - Valid Palindrome.c
#include <string.h>
#include <type.h>

bool isPalindrome(char* s) {
    int left, right;
    
    left = 0;
    right = strlen(s) - 1;
    
    while (left < right) {
        if (! (isalpha(s[left]) || isdigit(s[left]))) {
            ++left;
            continue;
        }
        
        if (! (isalpha(s[right]) || isdigit(s[right]))) {
            --right;
            continue;
        }
        
        if (tolower(s[left++]) != tolower(s[right--]))
            return 0;
    }
    
    return 1;
}

File: csharp/125.valid-palindrome.cs
/*
 * @lc app=leetcode id=125 lang=csharp
 *
 * [125] Valid Palindrome
 */
public class Solution {
    public bool IsPalindrome(string s) {
        int left = 0, right = s.Length - 1;
        while (left < right) {
            char lc = s[left];
            if (!Char.IsLetter(lc) && !Char.IsNumber(lc)) {
                ++left;
            } else {
                char rc = s[right];
                if (!Char.IsLetter(rc) && !Char.IsNumber(rc)) {
                    --right;
                } else {
                    if (Char.ToLower(lc) != Char.ToLower(rc)) {
                        return false;
                    }
                    ++left;
                    --right;
                }
            }
        }
        return true;
    }
}

